source/nodejs/tests/ui-tests/src/card-element-utils.ts

```diff
@@ -0,0 +1,347 @@
+    underlyingElement?: WebElement = undefined;
+
+    protected constructor() {}
+
+    elementWasFound(): boolean
+    {
+        return (this.underlyingElement !== undefined);
+    }
+}
+
+export abstract class Input extends Element {
+    protected constructor() {
+        super();
+    }
+
+    abstract setData(data: string): Promise<void>;
+}
+
+export class InputText extends Input {
+    private constructor(){
+        super();
+    }
+
+    static async getInputWithId(id: string): Promise<InputText>
+    {
+        let input = new InputText();
+        input.underlyingElement = await TestUtils.getInstance().getInput(id, "ac-textInput");
+        return input;
+    }
+
+    override async setData(data: string): Promise<void>
+    {
+        await this.underlyingElement?.click();
+        await this.underlyingElement?.sendKeys(data);
+    }
+
+    async isFocused(): Promise<boolean>
+    {
+        if (this.elementWasFound())
+        {
+            const inputId: string = await this.underlyingElement!.getAttribute("id");
+            const activeElementId: string = await TestUtils.getInstance().getDriver().switchTo().activeElement().getAttribute("id");
+
+            return (inputId === activeElementId);
+        }
+
+        return false;
+    }
+}
+
+export class InputDate extends Input {
+    private constructor(){
+        super();
+    }
+
+    static async getInputWithId(id: string): Promise<InputDate>
+    {
+        let input = new InputDate();
+        input.underlyingElement = await TestUtils.getInstance().getInput(id, "ac-dateInput");
+        return input;
+    }
+
+    override async setData(data: string): Promise<void> 
+    {
+        await this.underlyingElement?.click();
+        await this.underlyingElement?.sendKeys(data);
+    }
+}
+
+export class InputTime extends Input {
+    
+    private constructor(){
+        super();
+    }
+
+    static async getInputWithId(id: string): Promise<InputTime>
+    {
+        let input = new InputTime();
+        input.underlyingElement = await TestUtils.getInstance().getInput(id, "ac-timeInput");
+        return input;
+    }
+    
+    override async setData(data: string): Promise<void> {
+        await this.underlyingElement?.click();
+        await this.underlyingElement?.sendKeys(data);
+    }
+    
+}
+
+export class InputNumber extends Input {
+
+    private constructor(){
+        super();
+    }
+
+    static async getInputWithId(id: string): Promise<InputNumber>
+    {
+        let input = new InputNumber();
+        input.underlyingElement = await TestUtils.getInstance().getInput(id, "ac-numberInput");
+        return input;
+    }
+
+    override async setData(data: string): Promise<void> {
+        await this.underlyingElement?.click();
+        await this.underlyingElement?.sendKeys(data);
+    }
+}
+
+export class InputChoiceSet extends Input 
+{
+    isExpanded: boolean = false;
+    isMultiSelect: boolean = false;
+    underlyingElements: WebElement[];
```

Comment:
wondering if this should be renamed a bit to make it a little more clear that it's distinct from `underlyingElement`...